packages/integrations/test/pi-hole.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { GenericContainer } from "testcontainers";
+import { describe, expect, test } from "vitest";
+
+import { PiHoleIntegration } from "../src";
+
+describe("Pi-hole integration", () => {
+  test("getSummaryAsync should return summary from pi-hole", async () => {
+    // Arrange
+    const password = "12341234";
+    const apiKey = "3b1434980677dcf53fa8c4a611db3b1f0f88478790097515c0abb539102778b9";
+
+    const piholeContainer = await new GenericContainer("pihole/pihole:latest")
+      .withEnvironment({
+        WEBPASSWORD: password,
+      })
+      .withExposedPorts(80)
+      .start();
+
+    // We need to wait until pi hole has actually started and is ready to get requests
+    await sleep(5000);
```

Comment:
Isn't there a better way? (eg. healthchecks)
On fast systems this will be an artificial delay which will greatly add up if you run dozens of tests

Reply:
Yes there is a better way, I've added it now. Wasn't sure anymore how it has to be done and wanted to test if the test would run successfully on GitHub